File: src/main/java/be/zvz/kotlininside/value/ApiUrl.java
package be.zvz.kotlininside.value;

public final class ApiUrl {
    public static final String PC_WEB = Protocol.HTTPS + "gall.dcinside.com/";
    public static final String MOBILE_WEB = Protocol.HTTP + "m.dcinside.com/";
    public static final String APP_API = Protocol.HTTPS + "app.dcinside.com/api/";
    public static final String AUTH_API = Protocol.HTTPS + "dcid.dcinside.com/join/";
    public static final String MAIN_API = Protocol.HTTP + "json2.dcinside.com/";
    public static final String UPLOAD = Protocol.HTTPS + "upload.dcinside.com/";

    private static final class Protocol {
        private static final String HTTP = "http://";
        private static final String HTTPS = "https://";
    }

    public static final String REDIRECT = APP_API + "redirect.php";

    public static final class Article {
        public static final String LIST = APP_API + "gall_list_new.php";
        public static final String READ = APP_API + "gall_view_new.php";
        public static final String WRITE = UPLOAD + "_app_write_api.php";
        public static final String DELETE = APP_API + "gall_del.php";
        public static final String MODIFY = APP_API + "gall_modify.php";

        public static final String UPVOTE = APP_API + "_recommend_up.php";
        public static final String DOWNVOTE = APP_API + "_recommend_down.php";

        public static final String REPORT = MOBILE_WEB + "api/report.php";

        public static final String HIT_UPVOTE = APP_API + "hit_recommend";
    }

    public static final class Comment {
        public static final String OK = APP_API + "comment_ok.php";
        public static final String DELETE = APP_API + "comment_del.php";
        public static final String READ = APP_API + "comment_new.php";
    }

    public static final class DCCon {
        public static final String DCCON = APP_API + "dccon.php";
    }

    public static final class Gallery {
        public static final String MINOR_INFO = APP_API + "minor_info";
        public static final String MINOR_MANAGEMENT = PC_WEB + "mgallery/management/mobile";
        public static final String MINOR_MANAGER_REQUEST = APP_API + "_manager_request.php";
        public static final String MINOR_BLOCK_WEB = APP_API + "minor_avoid";
        public static final String MINOR_BLOCK_ADD = APP_API + "minor_avoidadd";
    }

    public static final class Search {
        public static final String SEARCH = APP_API + "_total_search.php";
    }

    public static final class Auth {
        public static final String LOGIN = AUTH_API + "mobile_app_login.php";
        public static final String APP_ID = AUTH_API + "mobile_app_key_verification_3rd.php";
        public static final String APP_CHECK = MAIN_API + "json0/app_check_A_rina_beta.php";
    }

    public static final class User {
        public static final String MY_GALL = APP_API + "mygall.php";
        public static final String MY_GALL_MODIFY = APP_API + "mygall_modify.php";
        public static final String MY_MANAGE_GALL_CHECK = APP_API + "mymanageGallChk";
        public static final String MY_MINI_JOIN_CHECK = APP_API + "myminijoinGallChk";
    }

    public static final class MiniGallery {
        public static final String JOIN = APP_API + "memberjoin";
        public static final String JOIN_OK = APP_API + "memberjoin_ok";
        public static final String QUIT = APP_API + "memberout_ok";
    }

    public static final class MainInfo {
        public static final String APP_MAIN = MAIN_API + "json3/main_content.php";
        public static final String GALLERY_RANKING = MAIN_API + "json3/ranking_gallery.php";
        public static final String MINOR_GALLERY_RANKING = MAIN_API + "json1/mgallmain/mgallery_ranking.php";
    }
}

File: src/main/kotlin/be/zvz/kotlininside/value/ApiUrl.kt
package be.zvz.kotlininside.value

object ApiUrl {
    const val PC_WEB = Protocol.HTTPS + "gall.dcinside.com/"
    const val MOBILE_WEB = Protocol.HTTP + "m.dcinside.com/"
    const val MOBILE_WEB_HTTPS = Protocol.HTTPS + "m.dcinside.com/"
    const val MOBILE_APP = Protocol.HTTPS + "app.dcinside.com/"
    const val APP_API = MOBILE_APP + "api/"
    const val AUTH_API = Protocol.HTTPS + "msign.dcinside.com/"
    const val MAIN_API = Protocol.HTTP + "json2.dcinside.com/"
    const val UPLOAD = Protocol.HTTPS + "upload.dcinside.com/"
    const val MOVIE_UPLOAD = Protocol.HTTPS + "m4up4.dcinside.com/"
    const val REDIRECT = APP_API + "redirect.php"

    private object Protocol {
        const val HTTP = "http://"
        const val HTTPS = "https://"
    }

    object Firebase {
        const val INSTALLATIONS =
            Protocol.HTTPS + "firebaseinstallations.googleapis.com/v1/projects/dcinside-b3f40/installations"
    }

    object PlayService {
        const val ANDROID_CLIENT = Protocol.HTTPS + "android.clients.google.com"
        const val REGISTER3 = "$ANDROID_CLIENT/c2dm/register3"
        const val CHECKIN = "$ANDROID_CLIENT/checkin"
    }

    object Upload {
        const val CHECK_UPLOAD_RESTRICTION = APP_API + "chk_upload_restriction"
        const val MOVIE = "$MOVIE_UPLOAD/movie_upload_v1.php"
    }

    object Article {
        const val LIST = APP_API + "gall_list_new.php"
        const val READ = APP_API + "gall_view_new.php"
        const val WRITE = UPLOAD + "_app_write_api.php"
        const val DELETE = APP_API + "gall_del.php"
        const val MODIFY = APP_API + "gall_modify.php"
        const val UPVOTE = APP_API + "_recommend_up.php"
        const val DOWNVOTE = APP_API + "_recommend_down.php"
        const val REPORT = MOBILE_WEB + "api/report.php"
        const val HIT_UPVOTE = APP_API + "hit_recommend"
        const val INSERT_MOVIE_INFO = MOBILE_APP + "movie/insert-mvinfo"
    }

    object Comment {
        const val OK = APP_API + "comment_ok.php"
        const val DELETE = APP_API + "comment_del.php"
        const val READ = APP_API + "comment_new.php"
    }

    object DCCon {
        const val DCCON = APP_API + "dccon.php"
    }

    object Gallery {
        const val MINOR_INFO = APP_API + "minor_info"
        const val MINOR_MANAGEMENT = PC_WEB + "mgallery/management/mobile"
        const val MINOR_NOMEMBER = MOBILE_WEB_HTTPS + "management/minor/nomember"
        const val MINOR_MANAGER_REQUEST = APP_API + "_manager_request.php"
        const val MINOR_BLOCK_WEB = APP_API + "minor_avoid"
        const val MINOR_BLOCK_ADD = APP_API + "minor_avoidadd"
    }

    object Search {
        const val SEARCH = APP_API + "_total_search.php"
    }

    object Auth {
        const val LOGIN = AUTH_API + "api/login"
        const val APP_ID = AUTH_API + "auth/mobile_app_verification"
        const val APP_CHECK = MAIN_API + "json0/app_check_A_rina_one.php"
    }

    object User {
        const val MY_GALL = APP_API + "mygall.php"
        const val MY_GALL_MODIFY = APP_API + "mygall_modify.php"
        const val MY_MANAGE_GALL_CHECK = APP_API + "mymanageGallChk"
        const val MY_MINI_JOIN_CHECK = APP_API + "myminijoinGallChk"
    }

    object MiniGallery {
        const val JOIN = APP_API + "memberjoin"
        const val JOIN_OK = APP_API + "memberjoin_ok"
        const val QUIT = APP_API + "memberout_ok"
    }

    object MainInfo {
        const val APP_MAIN = MAIN_API + "json3/main_content.php"
        const val GALLERY_RANKING = MAIN_API + "json3/ranking_gallery.php"
        const val MINOR_GALLERY_RANKING = MAIN_API + "json1/mgallmain/mgallery_ranking.php"
        const val MINI_GALLERY_RANKING = MAIN_API + "json1/migallmain/migallery_ranking.php"
    }
}
